mono/metadata/sgen-tarjan-bridge.c

```diff
@@ -398,6 +439,13 @@ dyn_array_ptr_contains (DynPtrArray *da, void *x)
 }
 
 static gboolean
+match_colors_estimate (DynPtrArray *a, DynPtrArray *b)
+{
+	return dyn_array_ptr_size (a) == dyn_array_ptr_size (b);
+}
+
+
+static gboolean
 match_colors (DynPtrArray *a, DynPtrArray *b)
```

Comment:
Isn't this function a little wasteful anyway? It looks like it checks to test that A and B contain the same items regardless of order. However, we could get away with just checking that the arrays are equal. If the arrays contained the same items in a different order, we wouldn't even get here, because the hashes would be different.